src/HackUnlock/model.js

```diff
@@ -0,0 +1,25 @@
+/* exported HUModelGlobal */
+
+const {GObject} = imports.gi;
+const {HBModelBase} = imports.hackyballs.model;
+
+const _propFlags = GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT;
+
+var HUModelGlobal = GObject.registerClass({
+    Properties: {
+        amplitude: GObject.ParamSpec.double(
+            'amplitude', 'amplitude', '',
+            _propFlags, 0.0, 20.0, 3.0),
+        frequency: GObject.ParamSpec.double(
+            'frequency', 'frequency', '',
+            _propFlags, 5.0, 30.0, 20.0),
+        phase: GObject.ParamSpec.double(
+            'phase', 'phase', '',
+            _propFlags, -2.0, 2.0, 0.0),
+    },
+}, class HUModelGlobal extends HBModelBase {
+    _init(props = {}) {
+        super._init(props);
+        this.bindProperties('com.endlessm.HackUnlock', '/com/endlessm/HackUnlock', {});
```

Comment:
I'm a bit confused, don't you need to pass the property map here? There should be at least some no?

Reply:
The base class assume that if no/empty map is provided then the properties names are the same locally and remotely, but same as my comment above, ill should be able to remove all of this with new Clippy API.

Reply:
Oh, I see. I'll wait for that to land then before making any further refactoring suggestions :-)